[Christabel Teo] Duke Increments

christabelteo opened the pull request.

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

christabelteo committed:
Completed Level-1

christabelteo committed:
Completed Level-2

christabelteo committed:
Completed Level-3

christabelteo committed:
Completed Level-4

christabelteo committed:
Completed Level-5

the task exceptions could be more precised :(

christabelteo committed:
Completed Level-6

donzo for week2

christabelteo committed:
Updating Switch-case formatting

christabelteo committed:
Added Java Documentation

christabelteo committed:
Creating DukeData class to handle data file

And also some minor adjustments to the variable names

christabelteo committed:
Complete Level-7

Might wanna edit to change the indexing of the data file to accomodate bigger indices

christabelteo committed:
Incomplete Level-8

IllegalArgumentException not yet debugged

christabelteo committed:
Completed Level-8

christabelteo committed:
Changing variable names

christabelteo committed:
Merge branch 'master' into branch-Level-8

christabelteo committed:
Merge pull request #2 from belbel98/branch-Level-8

Branch level 8 to master

christabelteo committed:
Extracted code for Ui class

Have yet to integrate the methods that need TaskList class to be added.

christabelteo committed:
Extracted code for TaskList class

Have yet to integrate the new TaskList methods into the Duke class

christabelteo committed:
Complete A-MoreOOP

christabelteo committed:
Revert "Complete A-MoreOOP"

This reverts commit ed5cc5eb093aac23a52d1ccc81b96b99c749f0ff.

christabelteo committed:
Revert "Revert "Complete A-MoreOOP""

This reverts commit d4d8afbf463252474ffabeb47e0d63474f766066.

christabelteo committed:
Completed JUnit Testing

Added tests for ToDo and DateTime classes

christabelteo committed:
Completed A-JavaDoc

christabelteo committed:
JavaDocs generated, plus minor adjustments

christabelteo committed:
Completed A-CodingStandard

corrected unclear variable names, and packing files into subpackages

christabelteo committed:
Merge pull request #3 from belbel98/branch-A-JavaDoc

Merge A-JavaDoc to master branch

christabelteo committed:
Update ToDo.java

christabelteo committed:
Merge branch 'master' into branch-A-CodingStandard

christabelteo committed:
Merge branch 'branch-A-CodingStandard' of https://github.com/belbel98/duke into branch-A-CodingStandard

christabelteo committed:
Merge pull request #4 from belbel98/branch-A-CodingStandard

Merging A-CodingStandard to master branch

christabelteo committed:
Remove unnecessary class files

christabelteo committed:
Completed Level-9

christabelteo committed:
Merge branch 'gradle'

christabelteo committed:
Comleted A-Gradle and A-CheckStyle

christabelteo committed:
Built a working prototype for GUI

Implemented up to gui/javafx tutorial 3

ShaunNgTX commented on src/JavaDocs/DateTime.html:
```diff
@@ -0,0 +1,324 @@
+<!DOCTYPE HTML>
```
There's a bunch of html files that I don't really understand whats going on. Could you kindly teach me if I am missing out anything? Why do you have all these html files in your duke project?

ShaunNgTX commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,65 @@
+        Duke theDuke = new Duke("/Users/StudyBuddy/Desktop/CS2103/iP/duke/src/main/data");
+        theDuke.run();
+    }
+}
```
I like how your duke file is so clean. I think that was what you meant when you told me to make my code more OOP. nice!

ShaunNgTX replied:
I like how your duke file is so clean. I think that was what you meant when you told me to make my code more OOP. nice!

ShaunNgTX commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,134 @@
+        while (sc.hasNext()) {
+            String cmdWord = sc.next(); // extract first only
+            String cmdLine = sc.nextLine(); // extract the rest of the line
+
```
I think this is a great way of parsing the line, at least much more efficient than my method of splitting the string

ShaunNgTX commented on src/test/java/ToDoTest.java:
```diff
@@ -0,0 +1,23 @@
+        todo.markAsDone();
+        assertEquals(icon, todo.getStatusIcon());
+    }
+}
```
Overall, i like how clean the code is. Each file has a specific function and the code is not cluttered in a single file. I actually learnt quite a bit from how you wrote your codes, and how you organise them! Great work! I am actually looking forward to seeing more of your coding style in the future!

ShaunNgTX commented on src/main/java/duke/command/DukeException.java:
```diff
@@ -0,0 +1,25 @@
+    public String getMessage() {
+        return this._message;
+    }
+}
```
This together with all the other exceptions that you made make up very comprehensive exception handling! Great work

ShaunNgTX commented on src/test/java/ToDoTest.java:
```diff
@@ -0,0 +1,23 @@
+        todo.markAsDone();
+        assertEquals(icon, todo.getStatusIcon());
+    }
+}
```
Overall, i like how clean the code is. Each file has a specific function and the code is not cluttered in a single file. I actually learnt quite a bit from how you wrote your codes, and how you organise them! Great work! I am actually looking forward to seeing more of your coding style in the future!

christabelteo replied:
> Each file has a specific function and the code is not cluttered in a single file.

Yes, i guess you can make use of this idea to create a more OOP structure for your project!

ShaunNgTX commented on src/JavaDocs/DateTime.html:
```diff
@@ -0,0 +1,324 @@
+<!DOCTYPE HTML>
```
There's a bunch of html files that I don't really understand whats going on. Could you kindly teach me if I am missing out anything? Why do you have all these html files in your duke project?

christabelteo replied:
Hmm all these html files are the html version of the JavaDocs generated. If you use IntelliJ, you can generate JavaDocs by going to Tools, and then click on "generate JavaDocs", it will open the html file in the browser and save it to your repo.

shawnkoh commented:
It's good practice to gitignore dynamic files, like javadocs and .jar files.
Had a shock when I saw that your PR has 56k delta haha

shawnkoh commented on src/main/java/duke/command/DukeException.java:
```diff
@@ -0,0 +1,25 @@
+ */
+public class DukeException extends Exception {
+    private final String _message = " OOPS!!! I'm sorry, but I don't know what that means :-(";
+    private String _error;
```
You should avoid using _ in variable names.
https://www.geeksforgeeks.org/using-_-underscore-variable-name-java/

shawnkoh commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,134 @@
+     * Creates a new Parser object with the given Duke program.
+     * @param parseDuke the Duke Program that will be handles by the Parser
+     */
+    public Parser(Duke parseDuke) {
```
It would be better if Parser was decoupled from Duke.

shawnkoh commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,134 @@
+     * The parse method handles all input and output from the Scanner,
+     * as well as their corresponding interactions through the Duke program.
+     */
+    public void parse() {
```
You can use parse(String) to decouple.

shawnkoh commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,134 @@
+
+            try {
+                switch (cmdWord) {
+                    case "list":
```
code style - case statements should not be indented and there should be a linebreak after every case.
Consider using https://plugins.jetbrains.com/plugin/1065-checkstyle-idea

shawnkoh commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,134 @@
+                        this._data.exit();
+                        System.out.println(_ui.showFarewell());
+                        break;
+                    case "done":
```
Consider enumerating your command words, or declaring them as constants in a Command (can check out my repo on how I did it)

Both ways are valid, depending on preference and use. See https://stackoverflow.com/questions/9969690/whats-the-advantage-of-a-java-enum-versus-a-class-with-public-static-final-fiel

shawnkoh commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,134 @@
+                }
+            } catch (DukeException e) {
+                System.err.println(Ui.addLines(e.getMessage()));
+            } catch (ToDoException e) {
```
There is no need to have one Exception class for every Task.
As much as possible, you should use standard Java exceptions and only use the custom DukeException when absolutely necessary. See https://github.com/nus-cs2103-AY1920S1/forum/issues/20

shawnkoh commented on src/main/java/duke/command/Ui.java:
```diff
@@ -0,0 +1,146 @@
+ * String representations of the output for the user.
+ */
+public class Ui {
+    private static final String LOGO = "____        _        \n"
```
Avoid using \n. Use System.lineSeperator() wherever possible.

shawnkoh commented on src/main/java/duke/data/DukeData.java:
```diff
@@ -0,0 +1,183 @@
+            // attach the file to FileWriter and BufferedWriter
+            this._fw = new FileWriter(this._saveFile);
+            this._bw = new BufferedWriter(this._fw);
+        } catch (IOException e) {
```
Should not catch this exception unless you help the user recover enter a new pathname and try again.

shawnkoh commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,55 @@
+     * @param desc a description of the Deadline Task
+     * @param date the date of the Deadline which is to be done by
+     */
+    public Deadline(String desc, String date) {
```
Would be better if you used description instead of desc.

shawnkoh commented on src/main/java/duke/task/DeadlineException.java:
```diff
@@ -0,0 +1,37 @@
+ * and prompts them to enter in the correct format.
+ */
+public class DeadlineException extends Exception{
+    private static String _noDesc = " OOPS!!! The description of a deadline cannot be empty.";
```
You should declare them as final too.

shawnkoh commented on src/main/java/duke/task/DeadlineException.java:
```diff
@@ -0,0 +1,37 @@
+     * @return  a string of error message for the exception found in the Deadline object.
+     */
+    public String getMessage() {
+        if (this._type == 1) {
```
Better to use enums than magic numbers.

shawnkoh commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,47 @@
+            this._dateTime = new DateTime(date);
+            this._date = this._dateTime.getDateTimeString();
+        } catch (ParseException e) {
+            System.err.println("Cant parse Date: " + date);
```
It'll be better if you returned a String with the feedback message instead of println, especially if you want to pass it to your GUI.
It's even better if you declare a CommandResult class.

shawnkoh commented on src/main/java/duke/task/EventException.java:
```diff
@@ -0,0 +1,39 @@
+ * and prompts them to enter in the correct format.
+ */
+public class EventException extends Exception {
+    private static String _noDesc = " OOPS!!! The description of an event cannot be empty.";
```
declare final.

shawnkoh commented on src/main/java/duke/task/EventException.java:
```diff
@@ -0,0 +1,39 @@
+    public String getMessage() {
+        if (this._type == 1) {
+            return _noDesc;
+        } else if (this._type == 2){
```
When using guard clauses, its not necessary to use else if because the if statement will exit the method.

shawnkoh commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,61 @@
+     * @return a String representation of the Task object
+     */
+    @Override
+    public abstract String toString();
```
It would be better if you defined toString() and use super.toString() in your Task subclasses to minimise code.

shawnkoh commented on src/main/java/duke/task/ToDoException.java:
```diff
@@ -0,0 +1,27 @@
+ * and prompts them to enter in the correct format.
+ */
+public class ToDoException extends Exception{
+    private final String _noDesc = " OOPS!!! The description of a todo cannot be empty.";
```
constants should be private static final

christabelteo committed:
Update .gitignore

Added JavaDocs file path, compiled and packaged files to .gitignore

christabelteo committed:
Level-10 in-progress

GUI is not working yet

christabelteo committed:
.gitignore fix

christabelteo committed:
Update .gitignore

christabelteo committed:
Revert "Level-10 in-progress"

This reverts commit 3fafb4d3d0a870fc47c68a3a2f9764a9cd8719a3.

christabelteo committed:
Implemented working GUI

Duke program not yet executable in the GUI

christabelteo committed:
Minor Adjustments to GUI

christabelteo committed:
Added Command classes

Decoupled duke from parser. These allows for a better handling of the commands so it can be supported by the GUI.

christabelteo committed:
Merge pull request #7 from belbel98/branch-Level-10

Completed Level-10

christabelteo committed:
Branch A-CodeQuality (#8)

* Amendment of JavaDocs

* Renamed variables to adhere to coding standard

christabelteo committed:
Add .txt files to .gitignore

christabelteo committed:
Add Extension D-Quotes

D-Quotes is an adaptation for D-Trivia extension in the available list of extension categories.

christabelteo committed:
Fix bugs for D-Quotes

christabelteo committed:
Complete A-UserGuide

christabelteo committed:
Set theme jekyll-theme-architect

christabelteo committed:
Released working v0.1.9

christabelteo committed:
Decoupled TaskList from DukeData (Storage)

christabelteo committed:
Minor adjustments

christabelteo committed:
Update Checkstyle Adjustments

christabelteo committed:
Fixed DukeData and DateTime issue

christabelteo committed:
Release final version v0.3.0
